Buddies/MyActivities/MyActivitiesVC.swift

```diff
@@ -30,11 +30,17 @@ class MyActivitiesVC: ActivityTableVC, UISearchBarDelegate {
     
     
     override func viewWillAppear(_ animated: Bool) {
+        //Super calls fetchAndLoadActivities
         super.viewWillAppear(animated)
         cancelUserListener = DataAccessor.instance.useLoggedInUser{ user in
+            guard let user = user else { return }
+            let oldUser = self.user
+            // If user changed
             self.user = user
+            if user.uid != oldUser?.uid {
```

Comment:
What does this accomplish? E.g. this should always return false, right? Since your UID will never change. Or is this mostly for the `nil` check?